Add introduction to Dapp Development Guide (#223)
onflow/flow: docs/content/guides/dapp-deployment/index.md

description: How to deploy an application on Flow
---

## Introduction

This guide covers the steps a developer should take to prepare their application for deployment
on Flow Mainnet.

The steps below present a framework that Flow dapp developers are encouraged to follow
when developing and preparing their applications for Mainnet. 

> Flow is still in its infancy, meaning that many best practices have yet to be defined. 

Reviewer: are still currently being defined maybe sounds better?

Author: Yep good call!

This guide aims to lay the foundation for secure and robust dapp development on Flow.

### Who created this guide?

These documents were compiled by the Flow core developers in collaboration with the wider development community. 
Suggestions for additions, changes or others improvements are gladly welcomed.

## The Road to Mainnet

**1. [Smart Contract Testing](./contract-testing)**

Smart contracts are the most important component of any dapp: 
they often manage user assets, application state and other vital pieces of data.
As such, smart contracts should be rigorously tested.

**2. [Deploy to Testnet](./testnet-deployment)**

Testnet is the perfect proving ground for any new application.
Learn how to use the Flow toolchain to deploy your dapp to Testnet.

**3. [Testnet Testing](./testnet-testing)**

Testnet is the closest you will get to the real thing.
In order to uncover bugs that may only appear on a real network,
you should thoroughly test your dapp on Testnet before considering a Mainnet release.

**4. [Deploy to Mainnet](./mainnet-deployment)**

> Deployment to Mainnet is restricted in the near term in order to ensure

Reviewer: I don't think we should say restricted maybe reserved for those who pass our review or something alonmg those lines?

description: How to deploy an application on Flow
---

## Introduction

This guide covers the steps a developer should take to prepare their application for deployment
on Flow Mainnet.

The steps below present a framework that Flow dapp developers are encouraged to follow
when developing and preparing their applications for Mainnet. 

> Flow is still in its infancy, meaning that many best practices are still being defined. 

Reviewer 2: ```suggestion
Flow is still very new, meaning that many best practices are still being defined. 
```
